src/storage-client/src/controller.rs

```diff
@@ -1651,83 +1647,7 @@ where
 
         Ok(update)
     }
-}
-
-/// State maintained about individual collections.
-#[derive(Debug)]
-pub struct CollectionState<T> {
```

Comment:
Is this code movement on purpose? If not can we minimize these moves to have a nicer `git blame` output in the future?

Reply:
Yes, this is purposeful (had multiple `impl` blocks that were essentially identical).